### ISSUE TITLE
Sometimes Brave is unresponsive

### ISSUE THREAD

**Issue by btlechowski** [brave/brave-ios#733]
### Steps to Reproduce 
Test case 1: smash tab button
  1. Install Brave
  2. Open Brave and immediately smash `tabs` button (bottom right corner)

Test case 2: Brave is unresponsive after opening a private tab and going back to normal tab
  1. Install and open Brave
  2. Tap `tabs` button (bottom right corner)
  3. Tap `Private`
4. Tap `+` to add provate tab
5. Tap `No` on `Private search with DuckDuckGo?` modal
6. tap `Not Now` on `Private means private` modal
7. Tap `tabs` button (bottom right corner)
8. Tap `Private`
 

**Actual result:** 
Brave is unresponsive to input.
Needed to force close it.

**Expected result:**
No freeze.

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
random (30% reproduction rate)

**Brave Version:** 
1.7.2 (19.1.15.15)
Also reproduced on 1.7.3 (19.1.14.19), but tested only once

**Device details:** 
iPhone SE (iOS 12.1.2)

**Comment by btlechowski** [comment]
Still reproducible 1.8 (19.01.25.19), iPhone SE (iOS 12.1.2)

cc @brave/legacy_qa

**Comment by kjozwiak** [comment]
Ran into the PB case that @btlechowski mentioned several times where the entire browser becomes unresponsive and you're forced to close/re-open. Mentioned this a few times in the `#ios` channel.

**Comment by kylehickinson** [comment]
cc @danishjafri88

**Comment by jhreis** [comment]
Let's get some additional info on this, and add information here before working on the actual fix.

**Comment by btlechowski** [comment]
Still reproducible 1.9 (19.03.27.19), iPhone SE (iOS 12.1.2).

The issue is easily reproducible using `Test case 1` from the description

**Comment by danishjafri88** [comment]
@jhreis fix in #963 all cases.

**Comment by kylehickinson** [comment]
Closing for now as this is stale
